rtmp-client/src/main/cpp/librtmp-jni.c

```diff
@@ -76,10 +76,12 @@ JNIEXPORT jint JNICALL Java_net_butterflytv_rtmp_1client_RtmpClient_nativeRead
     RTMP *rtmp = (RTMP *) rtmpPointer;
     if (rtmp == NULL) {
         throwIOException(env, "First call open function");
+        return 0;
```

Comment:
Am I right in thinking there should always be a return statement after a call to `throwIOException`? If so there are missing return statements in Java_net_butterflytv_rtmp_1client_RtmpClient_nativeWrite and Java_net_butterflytv_rtmp_1client_RtmpClient_nativePause as well. Can we fix those as part of this change, whilst we're doing this?

Reply:
Yes, control doesn't go back to Java after a throw() call. An explicit return is needed.

Return statements have been added in the nativeWrite and nativePause functions.